src/pages/iou/request/step/IOURequestStepDistance.js

```diff
@@ -150,17 +150,21 @@ function IOURequestStepDistance({
             }
 
             const newWaypoints = {};
+            let emptyWaypointIndex = -1;
             _.each(data, (waypoint, index) => {
                 newWaypoints[`waypoint${index}`] = lodashGet(waypoints, waypoint, {});
+                // Find waypoint that BECOMES empty after dragging
+                if (_.isEmpty(newWaypoints[`waypoint${index}`]) && !_.isEmpty(lodashGet(waypoints, `waypoint${index}`, {}))) {
+                    emptyWaypointIndex = index;
+                }
             });
 
             setOptimisticWaypoints(newWaypoints);
-            // eslint-disable-next-line rulesdir/no-thenable-actions-in-views
-            Transaction.updateWaypoints(transactionID, newWaypoints, true).then(() => {
+            Promise.all([Transaction.removeWaypoint(transaction, emptyWaypointIndex, true), Transaction.updateWaypoints(transactionID, newWaypoints, true)]).then(() => {
```

Comment:
The signature of `removeWaypoint` says it expects a string for the index, but it can be a number here, is that going to complicate things when we have to migrate this file to TS?